src/main/java/htsjdk/variant/vcf/VCFMetaDataLines.java

```diff
@@ -0,0 +1,298 @@
+        List<VCFContigHeaderLine> contigLines = new ArrayList<>();
+        mMetaData.values().stream()
+                .filter(hl -> hl.getKey().equals(VCFConstants.CONTIG_HEADER_KEY))
+                .forEach(hl -> contigLines.add((VCFContigHeaderLine) hl));
+        return Collections.unmodifiableList(contigLines);
+    }
+
+    //TODO: Is this useful ? It returns the first match for the given key, no matter how many
+    //TODO: there are. Should we deprecate it (and add a new one that returns a collection)
+    //TODO: or just change this one to return a collection ?
+    /**
+     * Get the VCFHeaderLine whose key equals key.  Returns null if no such line exists
+     * @param key
+     * @return
+     */
+    public VCFHeaderLine getMetaDataLine(final String key) {
+        return mMetaData.values().stream()
+                .filter(hl -> hl.getKey().equals(key))
+                .findFirst()
+                .orElseGet(()->null);
+    }
+
+    /**
+     * Returns the INFO HeaderLines in their original ordering
+     */
+    public Collection<VCFInfoHeaderLine> getInfoHeaderLines() {
+        return mMetaData.values().stream()
+                .filter(hl -> hl.getKey().equals(VCFConstants.INFO_HEADER_KEY))
+                .map(hl -> (VCFInfoHeaderLine) hl)
+                .collect(Collectors.toCollection(ArrayList::new));
+    }
+
+    /**
+     * Returns the FORMAT HeaderLines in their original ordering
+     */
+    public Collection<VCFFormatHeaderLine> getFormatHeaderLines() {
+        return mMetaData.values().stream()
+                .filter(hl -> hl.getKey().equals(VCFConstants.FORMAT_HEADER_KEY))
+                .map(hl -> (VCFFormatHeaderLine) hl)
+                .collect(Collectors.toCollection(ArrayList::new));
+    }
+
+    /**
+     * @param id the id of the requested header line
+     * @return the meta data line, or null if there is none
+     */
+    public VCFInfoHeaderLine getInfoHeaderLine(final String id) {
+        return (VCFInfoHeaderLine) mMetaData.get(makeKey(VCFConstants.INFO_HEADER_KEY, id));
+    }
+
+    /**
+     * @param id  the id of the requested header line
+     * @return the meta data line, or null if there is none
+     */
+    public VCFFormatHeaderLine getFormatHeaderLine(final String id) {
+        return (VCFFormatHeaderLine) mMetaData.get(makeKey(VCFConstants.FORMAT_HEADER_KEY, id));
+    }
+
+    /**
+     * @param id the id of the requested header line
+     * @return the meta data line, or null if there is none
+     */
+    public VCFFilterHeaderLine getFilterHeaderLine(final String id) {
+        return (VCFFilterHeaderLine) mMetaData.get(makeKey(VCFConstants.FILTER_HEADER_KEY, id));
+    }
+
+    //TODO: Is this useful ? It returns the first match for the given key, no matter how many
+    //TODO: there are. Should we deprecate it (and add a new one that returns a collection)
+    //TODO: or just change this one to return a collection ?
```

Comment:
I vote for a breaking change to return a Collection, because anyway this PR will introduce lots of backwards incompatible changes...